ticketutil/servicenow.py

```diff
@@ -0,0 +1,321 @@
+
+    def process_response(self, response):
+        """
+        Helper function to process Requests response or raise exception
+
+        :param response: response of the Requests query
+        """
+        try:
+            if (response.status_code != 201
+                and response.status_code != 200
+                ):
+                raise
+            result = response.json()['result']
+            return result
+        except:
+            logging.error('Headers: {0}'.format(response.headers))
+            logging.error('Error Response: {0}'.format(response.json()))
+
+    def api_get(self, url):
+        """
+        API GET wrapper
+
+        :param url: url containing REST API query
+        """
+        self.s.headers.update({'Content-Type': 'application/json'})
+        response = self.s.get(url)
+        logging.debug('GET request Status Code: {0}'
+                      .format(response.status_code))
+        return self.process_response(response)
+
+    def api_post(self, url, data):
+        """
+        API POST wrapper
+
+        :param url: url containing REST API query
+        :param text: text to be inserted into table record
+        """
+        self.s.headers.update({'Content-Type': 'application/json',
+                               'Accept': 'application/json'})
+        response = self.s.post(url, data=data)
+        logging.debug('POST request Status Code: {0}'
+                      .format(response.status_code))
+        return self.process_response(response)
+
+    def api_put(self, url, data):
+        """
+        API PUT wrapper
+        Updates table content.
+
+        :param url: url containing REST API query
+        :param text: text to be inserted into table record
+        """
+        self.s.headers.update({'Content-Type': 'application/json',
+                               'Accept': 'application/json'})
+        response = self.s.put(url, data=data)
+        logging.debug('PUT request Status Code: {0}'
+                      .format(response.status_code))
+        return self.process_response(response)
+
+    def get_sys_id(self):
+        """
+        Get sys_id query wrapper
+        TODO: what about security, encodeURIComponent in python?
+
+        :return: sys_id
+        """
+        url = self.rest_url + '?sysparm_query=GOTOnumber%3D'
+        url += self.ticket_id
+        result = self.api_get(url)
+        try:
+            return result[0]['sys_id']
+        except:
+            self.ticket_id = None
+            self.sys_id = None
+            logging.error("Failed to get sys_id")
+
+    def _generate_ticket_url(self):
+        """
+        Generates the ticket URL out of the url, project, and ticket_id.
+
+        :return: ticket_url: The URL of the ticket.
+        """
+        ticket_url = None
+
+        # If we are receiving a ticket_id, set ticket_url.
+        if self.ticket_id:
+            self.sys_id = self.get_sys_id() if not self.sys_id else self.sys_id
```

Comment:
It would be more pythonic to do away with `get_sys_id` and instead turn `sys_id` into a property with an internal object attribute `_sys_id`. Doing so would avoid a line like this and you can simply refer to `self.sys_id` where appropriate.